### PR TITLE
Run cargo fix and fix some clippy warnings

### PR DESCRIPTION
**Pull request by lzutao**

### CODE REVIEW

**Comment by eddyb** [comment]
The reason `try!` was used was so this kept working on Rust 1.6 or something like that - I remember testing it on those old versions back when I added the `v0` module.

**Comment by lzutao** [comment]
Based on above ci test, this crate can only be built with Rust 1.31.0 and onward.

**Comment by eddyb** [comment]
I guess I'll leave it to @alexcrichton.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        rust: [stable, beta, nightly]
+        rust: [stable, beta, nightly, 1.31.0, 1.32.0, 1.33.0, 1.34.0, 1.35.0, 1.36.0, 1.37.0]
```
Comment (alexcrichton) [inline .github/workflows/main.yml]: Mind removing these older verisons? I personally prefer to just test with stable/beta/nightly nowadays

Comment (lzutao) [reply .github/workflows/main.yml]: Done.

**Pull request merged**